### ISSUE TITLE
Class generic type and some class methods not inferred with Celery.Task

### ISSUE THREAD

**Issue by orenl** [python/mypy#18900]
**Bug Report**

Given a task that inhrits `celery.Task`, mypy fails to infer the type of celery.Task.
It also fails to infer at least one method, namely `before_start()` to be called by super().
Note that mypy does recognize at least one other method, namely `after_return()`.

**To Reproduce**

```python
from typing import Any
import celery

class C(celery.Task):
    def before_start(self, task_id: str, args: tuple[Any, ...], kwargs: dict[str, Any]) -> None:
        super().before_start(task_id, args, kwargs)
```

**Expected Behavior**

```
$ mypy a.py
Success: no issues found in 1 source file
```

**Actual Behavior**
```
$ mypy a.py
/tmp/a.py:4: error: Missing type parameters for generic type "Task"  [type-arg]
/tmp/a.py:6: error: "before_start" undefined in superclass  [misc]
Found 2 errors in 1 file (checked 1 source file)
```

**Your Environment**

celery 5.5.0
celery-types 0.22.0
mypy 1.15.0

python 3.12.3

```
[tool.mypy]
strict = true
plugins = ["pydantic.mypy"]
```

**Comment by A5rocks** [comment]
Mypy does not infer class generics based on inheritance. Additionally, it looks like the `celery-types` package only recently added `before_start` which is why mypy is not aware of it: https://github.com/sbdchd/celery-types/pull/178. They have not cut a release since.

**Comment by orenl** [comment]
Thank you.
